### PR TITLE
Stub tests migration from java driver

### PR DESCRIPTION
**Pull request by injectives**
Migrated tests:
- shouldBeAbleRunCypher -> test_should_read_successfully_using_write_session_run
- shouldSendMultipleBookmarks -> test_send_and_receive_multiple_bookmarks_write_tx
- shouldSendNullRoutingContextForBoltUri -> test_should_exclude_routing_context (optimization)
- shouldSendReadAccessModeInQueryMetadata -> test_should_read_successfully_using_read_session_run
- shouldNotSendWriteAccessModeInQueryMetadata -> test_should_read_successfully_using_write_session_run
- shouldSendCustomerUserAgentInHelloMessage -> test_should_send_custom_user_agent_using_write_session_run

### CODE REVIEW

File: tests/stub/bookmarks/test_bookmarks_v4.py [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 from nutkit.protocol import AuthorizationToken
 from tests.shared import TestkitTestCase
 from tests.stub.shared import StubServer
 
 
-# TODO: Tests for 3.5 (no support for PULL n)
-
 # Tests bookmarks from transaction
-class TestBookmarks(TestkitTestCase):
+class TestBookmarksV4(TestkitTestCase):
```
Comment (robsdedude) [inline tests/stub/bookmarks/test_bookmarks_v4.py:8]: Renaming classes changes the test id an can break pipelines of other drivers if they want to skip a test from this class.  
So far, I've tried to make the transition of drivers that don't have a 4.4 branch yet as smooth as possible by running my changes against their 4.3 branches locally and making sure that they pass.  
If you haven't done that, I'd recommend doing so.

File: tests/stub/bookmarks/test_bookmarks_v4.py [hunk 2]
```diff
@@ -55,7 +54,8 @@ def test_last_bookmark(self):
 
     def test_send_and_receive_bookmarks_read_tx(self):
         self._server.start(
-            path=self.script_path("send_and_receive_bookmark_read_tx.script")
+            path=self.script_path(self.scripts_path,
```
Comment (robsdedude) [inline tests/stub/bookmarks/test_bookmarks_v4.py]: Nitpicky me: this naming is a bit confusing. Maybe rename `self.scripts_path` to something like `self.version_folder` or whatever. Or just ignore that comment, it's not a deal-breaker.

File: tests/stub/routing/test_no_routing_v4x1.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+from tests.shared import (
+    get_dns_resolved_server_address,
+    get_driver_name,
+    TestkitTestCase,
+)
+from tests.stub.shared import StubServer
+from ._routing import get_extra_hello_props
+
+
+class NoRoutingV4x1(TestkitTestCase):
```
Comment (robsdedude) [inline tests/stub/routing/test_no_routing_v4x1.py:12]: See other comment about renaming classes.

File: tests/stub/routing/test_no_routing_v4x1.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+    # Checks that routing is disabled when URI is bolt, no routing in HELLO and
+    # no call to retrieve routing table. From bolt >= 4.1 the routing context
+    # is used to disable/enable server side routing.
+    def test_should_read_successfully_using_read_session_run(self):
+        # Driver is configured to talk to "routing" stub server
+        uri = "bolt://%s" % self._server.address
+        self._server.start(
+            path=self.script_path(self.scripts_path, "reader.script"),
```
Comment (robsdedude) [inline tests/stub/routing/test_no_routing_v4x1.py]: Nitpicky: naming

**Review by robsdedude — APPROVED** [review]
PR :100: + :100: 
See comments.

File: tests/stub/bookmarks/test_bookmarks_v4.py (hunk 2 above)
Comment (injectives) [reply tests/stub/bookmarks/test_bookmarks_v4.py]: Personally, the chosen name was fine to me, but I do not mind updating it.

File: tests/stub/bookmarks/test_bookmarks_v4.py (hunk 1 above)
Comment (injectives) [reply tests/stub/bookmarks/test_bookmarks_v4.py:8]: I opened a similar PR to 4.3 and it seems like we are ok on this occasion (java is expected to fail at the moment).

**Pull request merged**